### PR TITLE
geo: fix st_segmentize to not panic when passed NaN as param

### LINKED ISSUE #62741: gemfn: st_segmentize error `makeslice: cap out of range`
Reproduce with the following query:

```
SELECT st_segmentize('0106000040060000000103000040010000000C000000D7CEC6343490FDC10024D5F224AB9CC18EB7A97A1718EBC17041B7EE6ED3C441C12D2D43FEEBF5C160D230A528F3BC4164BD8F69E37AFF41F39571B76B5101C2424B633C2E22FCC1E8476FB45267FC41206BA820F6DFF4C1D0646F4CFF7FE84120D406BE229AF84100A20DC892FFCCC10C1D67D17306E741CC8042B10518E241DCB9D1DBE12EE2411A800A0E4BCFE5C1C0C0F803685EA1C118AE6627B1D9F841324456F27A6EF941C04BAC22C0E4F5C1823090B80861F9415005C9217ED9F641254C2D5CC9FBFDC1585DA1CB0E4BEB410C729B5375FAF64100AE380FA7F4EEC110CFE3A47708D141100A7913D00BEE4144E2445FC098E1C10051114A6CF1A44123FB382734D901C2D7CEC6343490FDC10024D5F224AB9CC18EB7A97A1718EBC101030000400100000008000000F8A4EF3EA83AE8C13BE1E79F0D2FF3C1E0E8F682009DC8C1F093181A8851E7C11F851094E1FC00C2C0A6875DE67FAD417AFB17685D4F01424E68F37DE32FF5C12CADFEBEB839FF41A019A3EF0E0BE941606FCE4509BFC84150307B9D622EFC41345ED4054C71D5C1B0CD1CB3E161CB413A1BF68103C0E4C1483E2BA3CAABF2C18AE39C3D16ADF141A4A9C0D8C44C0042E61A56C554BBEEC12C1B8904DC58DDC1DEF8D467AF670142F8A4EF3EA83AE8C13BE1E79F0D2FF3C1E0E8F682009DC8C10103000040010000000A000000AA9D7CFAB0E9FFC10088F32F21157341200AC97F7F35B7C1FBB88DDBAEA4F7C1BA364120079CFAC1268F74AC49FCF6419C088B167A35D8C180EE7ED6997DA9C1A4AE479B0AE7FE41F98ABEF85127F1C1E647C4591A9100C2EC0641C674E6EBC1B86F5417CEE7D2C1B4531E55F32DE0C11C3BC9A2650DFE41F6F03E90E8D5F441C063962F5F2EF6C16852C78D5F63EFC144B122168343F341864CE7347B5E01421B408951359EF3C1284BF9AC0F90E741AEE5A70EC8D5004278BFD2AE3BE70042289345F1A76BE6C1F6FDE1211FEE0042B8871A64551002C2AA9D7CFAB0E9FFC10088F32F21157341200AC97F7F35B7C1010300004001000000070000000A16ACDDA6A2FDC1D895D53DB2ADF3C10067B04F4D2DF241B0EC6DDFEC2DD7C159276D81383E02C20C4978E1C7EAEF418C89C61385CEE741E475B9152E7AE5C1FDF94B9BF90DF5C1403BD42BA964E54110DABBC17D82CCC1938375F95660F0C13AED0190FD09F441200DC20BCD36D0415CCB556F7749F5C1DCDAFEB43F75E1C1F06D16956B1ECDC178825119387502420A16ACDDA6A2FDC1D895D53DB2ADF3C10067B04F4D2DF2410103000040010000000B000000F880AC029382F5C1867CEBB7AC6AEEC18DBAD3D13B7101C2BA2DFFACCF90F941A5BFDF8BC15201C2F62AA7680AB6F3415C1814B012D10142183DAE73A70DFEC13757674C36BEF1C15029576AF7A5FB4116B95ECD0129EAC168664CB8998F014256D92A8F523AFA419C9D61B1F23CE8C1D6860C127D3DF2413EF757402555F44144C242B27F6AE1C1D09E5870F9C4D3C120F3F297A1C4D641FAE85FF03459FA41340EEFB7F3FCED416846FBE665DDD04150E76C1F11CED54135BD109CD170FDC1C62C3DCEBBBB00C2504498FD086AC4419AD618706CB600C29EC30651E6A0EDC170B56ADCB4CBCEC158ED7EA9FBFDD641F880AC029382F5C1867CEBB7AC6AEEC18DBAD3D13B7101C20103000040010000000600000074874EE65F76F941DB3EC16383AFFDC108411DE301ACEA412C1E64DE4076F841844D34A49783E941E8BA1B7EC733D1C1F872D24C8783FB410CD9449A48D9F5410401A5142478E1410A47A4189744E2C12061930A672CFB4184B433781C01FD41BC7D6926AE35FAC1806ABF26C9FCFE411E7BBDA9B145F44174874EE65F76F941DB3EC16383AFFDC108411DE301ACEA41':::GEOMETRY, 'NaN':::FLOAT8);
```

```
ERROR: internal error: runtime error: makeslice: cap out of range
SQLSTATE: XX000
DETAIL: stack trace:
github.com/cockroachdb/cockroach/pkg/util/errorutil/catch.go:29: ShouldCatch()
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:168: func1()
runtime/panic.go:969: gopanic()
runtime/slice.go:31: panicmakeslicecap()
runtime/slice.go:95: makeslice()
github.com/cockroachdb/cockroach/pkg/geo/geomfn/segmentize.go:78: segmentizeCoords()
github.com/cockroachdb/cockroach/pkg/geo/geosegmentize/geosegmentize.go:71: Segmentize()
github.com/cockroachdb/cockroach/pkg/geo/geosegmentize/geosegmentize.go:86: Segmentize()
github.com/cockroachdb/cockroach/pkg/geo/geosegmentize/geosegmentize.go:99: Segmentize()
github.com/cockroachdb/cockroach/pkg/geo/geomfn/segmentize.go:40: Segmentize()
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/geo_builtins.go:4952: func466()
github.com/cockroachdb/cockroach/pkg/sql/sem/tree/eval.go:4024: Eval()
github.com/cockroachdb/cockroach/pkg/sql/opt/norm/fold_constants_funcs.go:576: FoldFunction()
```

cc @otan @andyyang890

### PR DESCRIPTION
**Pull request by andyyang890**
Fixes #62741.

Release note: None

### CODE REVIEW

File: pkg/geo/geogfn/segmentize.go [hunk 1]
```diff
@@ -25,6 +25,9 @@ import (
 // This works by dividing each segment by a power of 2 to find the
 // smallest power less than or equal to the segmentMaxLength.
 func Segmentize(geography geo.Geography, segmentMaxLength float64) (geo.Geography, error) {
+	if math.IsNaN(segmentMaxLength) || math.IsInf(segmentMaxLength, 1) {
```
Comment (otan) [inline pkg/geo/geogfn/segmentize.go:28]: nit: `, 1 /* sign */`

File: pkg/geo/geomfn/segmentize.go [hunk 2]
```diff
@@ -26,6 +26,9 @@ import (
 // between given two-points such that each segment has length less
 // than or equal to given maximum segment length.
 func Segmentize(g geo.Geometry, segmentMaxLength float64) (geo.Geometry, error) {
+	if math.IsNaN(segmentMaxLength) || math.IsInf(segmentMaxLength, 1) {
```
Comment (otan) [inline pkg/geo/geomfn/segmentize.go:29]: same nit here

File: pkg/geo/geogfn/segmentize_test.go [hunk 3]
```diff
@@ -100,6 +101,26 @@ func TestSegmentize(t *testing.T) {
...
 			expectedWKT:      "LINESTRING Z (0 0 25,0 1 0,0.49878052093921765 2.0003038990352664 25,0.9981696941692514 3.0004561476391296 50,1.4984735304805308 4.000380457593079 75,2 5 100)",
 		},
+		{
+			wkt:              "LINESTRING(0 0, 1 1)",
+			maxSegmentLength: math.NaN(),
+			expectedWKT:      "LINESTRING(0 0, 1 1)",
+		},
+		{
+			wkt:              "LINESTRING M (0 0 0, 1 1 1)",
+			maxSegmentLength: math.Sqrt(-1),
```
Comment (otan) [inline pkg/geo/geogfn/segmentize_test.go:111]: you monster

**Review by otan — APPROVED** [review]

**Comment by andyyang890** [comment]
bors r=otan

**Pull request merged**